src/tiledb/cloud/vector_search/file_ingestion.py

```diff
@@ -0,0 +1,354 @@
+    :param index_uri: URI of the vector index to load files to.
+    :param file_name: Name of the file to be loaded.
+    :param acn: Access Credentials Name (ACN) registered in TileDB Cloud (ARN type),
+        defaults to None.
+    :param config: config dictionary, defaults to None.
+    :param namespace: TileDB-Cloud namespace, defaults to None.
+    :param verbose: verbose logging, defaults to False.
+    :param trace_id: trace ID for logging, defaults to None.
+    # Index creation params
+    :param create_index: If true, creates a new vector search index.
+    :param index_type: Vector search index type ("FLAT", "IVF_FLAT").
+    :param index_creation_kwargs: Arguments to be passed to the index creation method
+    # DirectoryTextReader params.
+    :param include: File pattern to iclude relative to `file_dir_uri`. By default set
+        to include all files.
+    :param exclude: File patterns to exclude relative to `file_dir_uri`. By default
+        set to ignore all hidden files.
+    :param suffixes: Provide to keep only files with these suffixes
+        Useful when wanting to keep files with different suffixes
+        Suffixes must include the dot, e.g. ".txt"
+    :param max_files: Maximum number of files to include.
+    :param text_splitter_kwargs: Arguments for the splitter class.
+    # SentenceTransformersEmbedding params.
+    :param model_name_or_path: Huggingface SentenceTransformer model name or path
+    # Index update params.
+    :param index_timestamp: Timestamp to add index updates at.
+    :param workers: If `embeddings_generation_mode=BATCH` this is the number of
+        distributed workers to be used.
+    :param worker_resources: If `embeddings_generation_mode=BATCH` this can be used
+        to specify the worker resources.
+    :param worker_image: If `embeddings_generation_mode=BATCH` this can be used
+        to specify the worker Docker image.
+    :param extra_worker_modules: If `embeddings_generation_mode=BATCH` this can be used
+        to install extra pip package to the image.
+    :param driver_resources: If `embeddings_generation_driver_mode=BATCH` this can be
+        used to specify the driver resources.
+    :param driver_image: If `embeddings_generation_driver_mode=BATCH` this can be used
+        to specify the driver Docker image.
+    :param extra_driver_modules: If `embeddings_generation_driver_mode=BATCH` this can
+        be used to install extra pip package to the image.
+    :param max_tasks_per_stage: Number of maximum udf tasks per computation stage.
+    :param embeddings_generation_mode: TaskGraph execution mode for embeddings
+        generation.
+    :param embeddings_generation_driver_mode: TaskGraph execution mode for the ingestion
+        driver.
+    :param vector_indexing_mode: TaskGraph execution mode for the vector indexing.
+    :param index_update_kwargs: Extra arguments to pass to the index update job.
+    """
+
+    def ingest_files_udf(
```

Comment:
Since this is a purely-internal function and the parent function is guaranteed to pass in all values, the default values are not needed here.

Reply:
Removed